specs/0111-redesign-profile-page.md

```diff
@@ -0,0 +1,108 @@
+
+# Feature explanation
+
+See the [mockup](#mockup) for the overall view. Each of the following sections provides more details on different parts of the design.
+
+## Latest activity
+
+*Latest activity* is displayed in the sidebar. A user is considered active when they either submit a suggestion or review someone else’s work. It’s not considered activity:
+* Logging in to Pontoon.
+* Writing a comment.
+
+## Approval ratio
+
+Displayed in a graph, it’s the ratio between the number of translations approved over the total number of translations submitted, excluding self-approved translations (either submitted directly as translations, or provided as suggestions and later approved by the translator).
+
+Examples:
+* If a person submits 100 translations, but 20 are rejected, the approval ratio will be 80%.
+* If a person submits 100 translations, 20 are rejected, 20 are self-approved, the approval ratio will be 75% (60 over 80).
+
+The graph displays both the selected month’s ratio and the average for the previous 12 months.
+
+## Self-approval ratio
+
+Displayed in a graph, it’s the ratio between the number of translations submitted directly — or self-approved after submitting them as suggestions — over the total number of translations submitted. This data point is meaningful only for users with translation rights (translators, locale managers), as it will always be 0 for contributors.
+
+Example: if a person has 100 approved translations, but 60 are self-approved, the self-approval ratio will be 60%.
+
+The graph displays both the selected month’s ratio and the average for the previous 12 months.
+
+## Contribution graph
+
+The contribution graph displays data over time about the following actions:
+* `Submissions`: translations submitted, including both suggestions and direct translations.
+* `Reviews performed`: reviews performed on pending suggestions.
+* `Reviews received`: reviews received from another translator.
+* `Submissions and reviews`: submissions + reviews performed.
+* `All activities`: submissions + reviews performed + reviews received.
+
+Only one data point is displayed at a time, and it can be selected using the dropdown list above the graph. The default view is `All activities`, since it provides a good picture of the user’s activity in Pontoon.
+
+Each square in the graph represents a day, with color changing depending on the amount of contributions. It’s possible to click on a column to select a specific week, or change the year currently displayed using the links in the rightmost column.
+
+When selecting a specific time range, the data displayed below the graph is updated accordingly. This doesn’t affect the approval and self-approval graphs.
+
+## Other data in the sidebar
+
+The sidebar displays contact information for the contributor:
+* Name.
+* Description.
```

Comment:
This wasn't part of the proposal, but I noticed the mock-up had name, nick, and description.

Do we want to support also a nickname (didn't add it yet) and a description? It feels like a description would be taking away quite a bit of space.

Adding a nickname on top of the name would probably complicate the comments code, but I can see it as beneficial, as quoting some of the names is particularly complicated :-(

Reply:
I thought username (nickname) would be used for the vanity URL outlined below.

No strong thoughts on Description. It could add a personal touch or a summary if the page was used as a CV. It should be limited to e.g. 50 characters.

Reply:
> I thought username (nickname) would be used for the vanity URL outlined below.

I think it might be useful to have a nickname, especially if we support mentions in comments, but I would keep it separate from the vanity URL (e.g. I might want to use my full name for that, vs something silly in the nickname).

Reply:
It works for most platforms, though? I'd rather start small (nickname == vanity URL) and add a separate field in a followup if there's interest from the users. It's an optional field, which rarely get filled out.

Reply:
At that point, I think it makes more sense to only support the field which was originally part of the specs (vanity URL), instead of conflating the two. Splitting them up later would only create issues.

Also, properly supporting a nickname would require more work, and there's already plenty in this spec.